ZenToDesk.rb

```diff
@@ -3,105 +3,77 @@
 require 'rexml/document'
 require 'net/http'
 require 'json'
-include REXML
 
 #variables
 xmlfile = File.new(".xml")
-xmldoc = Document.new(xmlfile)
-root = xmldoc.root
-externalId = 0
-subject = ""
-createdAt = ""
-resolvedAt = ""
-description = ""
-count = 1
-totals = []
 
-#methods
-#create case object and send to Desk
-def CreateCase(subject, createdAt, resolvedAt, description, count, externalId)
-  uri = URI('https://yoursite.desk.com/api/v2/cases') # POST URI
-  req = Net::HTTP::Post.new(uri.path, {'Content-Type' => 'application/json'}) #set Post object (uri and content type header)
-  req.basic_auth '', '' #set Post object (auth)
-
-  #set Post object body && convert to json (contents of ticket)
-  req.body = {type: "email", external_id: "#{externalId}", subject: "#{subject}", priority: 4, status: "open", labels: ["archive"], created_at: "#{createdAt}", resolved_at: "#{resolvedAt}",message: {
-    direction: "in", subject: "#{subject}",body: "#{description}", to: "", from: "", created_at: "#{createdAt}"}}.to_json
-
-  #send the request
-  res = Net::HTTP.start(uri.hostname, uri.port,
-    :use_ssl => uri.scheme == 'https') do |http|
-      http.request(req)
+MigratesCases = Struct.new(:file) do
+  def parse_xml(doc)
+    doc.elements.to_a('tickets/ticket').each_slice(500) do |batch|
+      batch.each do |node|
+        parse_node(node)
+      end
+      sleep(60) # 500 API req/s
     end
-  if res.is_a?(Net::HTTPSuccess)
-    puts "Case Created!"
-    f = File.open("successLog.txt","a")
-    f.write("#{res.body}\n")
-    f.close
-    return true
-  else
-    puts "Oops! Case not created."
-    #error logging
-    f = File.open("errorLog.txt","a")
```

Comment:
If you want to log to a file, open it once and close it once, instead of opening/closing every time you log something. Opening/closing the file each time takes more time that writing to the file itself.